File: PlaygroundE/Assets/Scripts/Old/Stats.js
#pragma strict


    var max_health : float;
    var cur_health : float;
    var healthBar: GameObject;

    function Start()
    {
        setHealthBar();
    }

    function Update()
    {
        cur_health -= 10f * Time.deltaTime;
        setHealthBar();
        if(cur_health < 1)
        {
            Destroy(this.gameObject);
        }
    }

    function dec(input : float)
    {
        cur_health -= input;
    }

    function inc(input : float)
    {
        cur_health += input;
    }

    function setHealthBar()
    {
        var myHealth = cur_health / max_health;
        if (myHealth < 0) { myHealth = 0; }
        if (myHealth > 1) { myHealth = 1; }
        healthBar.transform.localScale = new Vector3(myHealth, healthBar.transform.localScale.y, healthBar.transform.localScale.z);
    }

File: PlaygroundE/Assets/Scripts/Stats.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.Networking;
using UnityEngine.SceneManagement;

public class Stats : NetworkBehaviour {

    public float max_health;
    [SyncVar(hook = "OnChangeHealth")]public float cur_health;
    public GameObject healthBar;
    private float timer = 0f;
    private GameObject NetPreb;
    public GameObject Guide;

    void Start()
    {
        //max_health = 100;
        //cur_health = 100;
        OnChangeHealth(cur_health);
        NetPreb = GameObject.Find("Network");
    }

    void Update()
    {
        if (!isLocalPlayer)
        {
            return;
        }
        if (timer > 3f)
        {
            this.Damage(1.0f);
            timer = 0f;
        }else{
            timer += Time.deltaTime;
        }
    }

    public void Damage(float input) 
    {
        if (!isServer)
        {
            return;
        }
        cur_health -= input;
    }

    void OnChangeHealth(float health)
    {
        if (health < 0)
        {
            if (isLocalPlayer)
            {
                GameObject guide = Instantiate(Guide) as GameObject;
                guide.transform.SetParent(this.transform.parent, false);
                NetPreb.GetComponentInChildren<NewHUD>().manager.StopHost();
                StartCoroutine(Restart());
            }
            if (isServer)
            {
                Destroy(this.gameObject);
            }
        }
        else
        {
            var myHealth = health / max_health;
            if (myHealth < 0) { myHealth = 0; }
            if (myHealth > 1) { myHealth = 1; }
            healthBar.transform.localScale = new Vector3(myHealth, healthBar.transform.localScale.y, healthBar.transform.localScale.z);
        }
    }
    IEnumerator Restart() {
        yield return new WaitForSeconds(2f);
        SceneManager.LoadScene("Main");
    }

}
